src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts

```diff
@@ -1019,7 +1019,14 @@ export class StoredFileWorkingCopy<M extends IStoredFileWorkingCopyModel> extend
 
 					// Delegate to working copy model save method if any
 					if (typeof resolvedFileWorkingCopy.model.save === 'function') {
-						stat = await resolvedFileWorkingCopy.model.save(writeFileOptions, saveCancellation.token);
+						const result = await raceCancellation(resolvedFileWorkingCopy.model.save(writeFileOptions, saveCancellation.token), saveCancellation.token);
```

Comment:
this is the wrong fix, we need to wait for the result to make sure the file doesn't get updated anyway